src/SWP/Bundle/ContentBundle/Factory/ArticleFactory.php

```diff
@@ -74,8 +76,13 @@ public function create()
      */
     public function createFromPackage(PackageInterface $package)
     {
+        /** @var ArticleInterface $article */
         $article = $this->create();
-        $article->setParentDocument($this->articleProvider->getParent($this->contentRelativePath));
+
+        if ($article instanceof HierarchyInterface) {
```

Comment:
we should go with PHPCR in this bundle imo too. I will simply extend `ArticleInterface` by `HierarchyInterface` here so that if statement won't be needed.